src/resource/createContainerInContainer.ts

```diff
@@ -0,0 +1,82 @@
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+// PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+// HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+// SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+import {
+  UrlString,
+  createContainerInContainer as coreCreateContainerInContainer,
+  WithResourceInfo,
+  SolidDataset,
+} from "@inrupt/solid-client";
+import { VerifiableCredential } from "@inrupt/solid-client-vc";
+import { fetchWithVc } from "../fetch";
+import { FetchOptions } from "../type/FetchOptions";
+
+/**
+ * Create an empty Container inside the Container at the given URL.
+ *
+ * Throws an error if creating the Container failed, e.g. because the current user does not have
+ * permissions to.
+ *
+ * The Container in which to create the new Container should itself already exist.
+ *
+ * This function is primarily useful if the current user does not have access to change existing files in
+ * a Container, but is allowed to add new files; in other words, they have Append, but not Write
+ * access to a Container. This is useful in situations where someone wants to allow others to,
+ * for example, send notifications to their Pod, but not to view or delete existing notifications.
+ * You can pass a suggestion for the new Resource's name, but the server may decide to give it
+ * another name — for example, if a Resource with that name already exists inside the given
+ * Container.
+ * If the user does have access to write directly to a given location, [[createContainerAt]]
+ * will do the job just fine, and does not require the parent Container to exist in advance.
+ *
+ * @param containerUrl URL of the Container in which the empty Container is to be created.
+ * @param options Optional parameter `options.fetch`: An alternative `fetch` function to make the HTTP request, compatible with the browser-native [fetch API](https://developer.mozilla.org/docs/Web/API/WindowOrWorkerGlobalScope/fetch#parameters). `options.slugSuggestion` accepts a string for your new Container's name.
+ * @returns A promise that resolves to a SolidDataset with ResourceInfo if successful, and that
+ * rejects otherwise.
+ * @since 0.2.0
+ */
+
+interface SaveInContainerOptions extends FetchOptions {
+  slugSuggestion?: string;
+}
+
+export async function createContainerInContainer(
+  containerUrl: UrlString,
+  accessGrant: VerifiableCredential,
+  options?: SaveInContainerOptions
+): Promise<SolidDataset & WithResourceInfo> {
+  const fetchOptions: FetchOptions = {};
+
+  if (options && options.fetch) {
+    fetchOptions.fetch = options.fetch;
+  }
+
+  const authenticatedFetch = await fetchWithVc(
+    containerUrl,
+    accessGrant,
+    fetchOptions
+  );
+
+  return await coreCreateContainerInContainer(containerUrl, {
+    ...options,
```

Comment:
Best to build up the options to pass along explicitly, as is done elsewhere

Reply:
Eee I still don't like this. It feels like bloat. What is the benefit again? If I am a developer I can look at the source code and explicitly see what options are available to me?

Reply:
it's mostly about providing a layer of isolation, as `options` here is not exactly the options that `coreCreateContainerInContainer` accepts, if we add an option here, what would happen is that it'd be exposed to `coreCreateContainerInContainer` even if that method doesn't accept that option.